src/k8s/pkg/component/gateway.go

```diff
@@ -0,0 +1,85 @@
+package component
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils/k8s"
+)
+
+func EnableGatewayComponent(s snap.Snap) error {
+	manager, err := NewManager(s)
+	if err != nil {
+		return fmt.Errorf("failed to get component manager: %w", err)
+	}
+
+	var values map[string]any = nil
+
+	err = manager.Enable("gateway", values)
+	if err != nil {
+		return fmt.Errorf("failed to enable gateway component: %w", err)
+	}
+
+	manager, err = NewManager(s)
+	if err != nil {
+		return fmt.Errorf("failed to get component manager: %w", err)
+	}
```

Comment:
do we need a second manager instance?

Reply:
Yes, this is related to Helm capabilities( as an example ` .Capabilities.APIVersions`) being fetched at `Init` time. If we don't recreate the manager instance, the cilium chart doesn't install the `GatewayClass` see [here](https://github.com/cilium/cilium/blob/39a474967deb6e04f09076433b0e5af7646fbabc/install/kubernetes/cilium/templates/cilium-gateway-api-class.yaml#L2)

Reply:
Ah, thanks. OK, fine with this short-term, let's not block the PR, we'll return to this in the future.